### PR TITLE
Updates 'clear search'  icon whenever filters are changed

### PR DESCRIPTION
**Pull request by DivyaPrabhakar**
References #3187

### CODE REVIEW

File: src/js/jsx/search/SearchBar.jsx [hunk 1]
```diff
@@ -194,6 +194,12 @@ define(function (require, exports, module) {
             var updatedFilter = id ? _.uniq(this.state.filter.concat(filterValues)) : [],
                 filterIcon = id && this.getFlux().store("search").getSVGClass(updatedFilter);
             
+            var nextHasInputValue = id !== null;
+            
+            if (nextHasInputValue !== this.state.hasInputValue) {
+                this.setState({ hasInputValue: nextHasInputValue });
```
Comment (volfied) [inline src/js/jsx/search/SearchBar.jsx]: Instead of calling setState here, can you add this key-value pair to the setState call that's made below?

**Comment by DivyaPrabhakar** [comment]
@volfied I made the change but I feel the `if` statement is a bit redundant. Let me know what you think and also, can you explain to me why one call to `setState`, instead of two, is better in this case?

**Comment by volfied** [comment]
Since this component does not have a `shouldComponentUpdate`, any calls you make to `setState` will cause a re-render. React usually tries to collect multiple `setState` calls together. In this case though, I'd like a single `setState` call for cleanliness. 

I would call `setState` with all three keys in both cases, and pass the same value. If you really care about the performance, you can implement a `shouldComponentUpdate` method and check the prevalent props and state variables to decide whether component should render or not. Youc an find examples of this in most of our React components.

**Comment by DivyaPrabhakar** [comment]
@volfied Got it, thanks!

**Comment by volfied** [comment]
:+1:

**Pull request merged**